src/acme/ui/CustomerListUI.java

```diff
@@ -0,0 +1,319 @@
+            {testData[2].getCustomerNumber(), testData[2].getName(), testData[2].getIntersection().getIntersectionName(), testData[2].isActive(),'\u270E' + " "+ testData[2].getName()},
+            {testData[3].getCustomerNumber(), testData[3].getName(), testData[3].getIntersection().getIntersectionName(), testData[3].isActive(),'\u270E' + " "+ testData[3].getName()},
+            {testData[4].getCustomerNumber(), testData[4].getName(), testData[4].getIntersection().getIntersectionName(), testData[4].isActive(),'\u270E' + " "+ testData[4].getName()}
+        },
+        new String [] {
+            "Number", "Name", "Location", "Active", "Edit"
+        }
+    ) {
+        Class[] types = new Class [] {
+            java.lang.String.class, java.lang.String.class, java.lang.String.class, java.lang.Boolean.class, javax.swing.JButton.class
+        };
+        boolean[] canEdit = new boolean [] {
+            //need last one true so can click button
+            false, false, false, false, true
+        };
+
+        public Class getColumnClass(int columnIndex) {
+            return types [columnIndex];
+        }
+
+        public boolean isCellEditable(int rowIndex, int columnIndex) {
+            return canEdit [columnIndex];
+        }
+    });
+    
+    //needed for buttons in table
+    TableColumn editCol = jTable1.getColumnModel().getColumn(4);
+    JButton editButton = new JButton("edit");
+    editCol.setCellRenderer(new ButtonRenderer());;
+    editCol.setCellEditor(new ButtonEditor(new JTextField()));
+  
+    //more table properties
+    jTable1.setColumnSelectionAllowed(false);
+    jTable1.getTableHeader().setFont(new Font("Lucida Grande", 0, 18)); 
+    jTable1.setRowHeight(24);
+    jTable1.setRowMargin(2);
+    jTable1.setSelectionBackground(new Color(93, 184, 202));
+    jTable1.setSelectionForeground(new java.awt.Color(0, 0, 0));
+    jTable1.setShowGrid(true);
+    jScrollPane1.setViewportView(jTable1);
+    jScrollPane1.setBounds(25, 75, 600, 120);
+    add(jScrollPane1);
+    
+    //new customer button
+    JButton btnNewCustomer = new JButton("New Customer");
+    btnNewCustomer.setBounds(25, 210, 120, 40);
+    btnNewCustomer.setBackground(new Color(93, 184, 202));
+    btnNewCustomer.setOpaque(true);
+    btnNewCustomer.addActionListener(new ActionListener() {
+      public void actionPerformed(ActionEvent e) {
+        goToNewCustomerPage();
+      }
+    });
+    add(btnNewCustomer);
+  }
+  
+  public static void main(String [] args) {
+    AcmeUI acme = new AcmeUI();
+    acme.setPanel(new CustomerListUI());
+    acme.getCompany().setCurrentUser(new User());
+}
+  
+//navigation methods-----------------------------
+ public void goToEditPage(String customerName)
+ { 
+   customerName = customerName.substring(2);
+   System.out.println(customerName);
+   //go to customer page
+ }
+ 
+ public void goToNewCustomerPage()
+ {
+   //go to new customer page
```

Comment:
same as above just send in null for a new customer

Reply:
done